target/linux/generic/hack-6.12/736-net-phy-realtek-add-rtl8251l-rtl8254b-rtl8261be-rtl8261n.patch

```diff
@@ -0,0 +1,704 @@
++		return 0;
++
++	int data = phy_read_mmd(phydev, MDIO_MMD_VEND1, 0x103);
++	if (data < 0)
++		return 0;
++
++	if (data != 0x8251)
++		return 0;
++
++	return 1;
++}
++
++static int rtl8254b_match_phy_device(struct phy_device *phydev,
++				     const struct phy_driver *phydrv)
++{
++	if (!rtlgen_is_c45_match(phydev, PHYID_RTL8264B, true))
++		return 0;
++
++	if (!phydev->mdio.bus->read_c45)
++		return 0;
++
++	int data = phy_read_mmd(phydev, MDIO_MMD_VEND1, 0x103);
++	if (data < 0)
++		return 0;
++
++	if (data != 0x8254)
++		return 0;
++
++	return 1;
++}
++
++static int rtl8261be_match_phy_device(struct phy_device *phydev,
++				      const struct phy_driver *phydrv)
++{
++	if (!rtlgen_is_c45_match(phydev, PHYID_RTL8261N, true))
++		return 0;
++
++	if (!phydev->mdio.bus->read_c45)
++		return 0;
++
++	int data = phy_read_mmd(phydev, MDIO_MMD_VEND1, 0x103);
++	if (data < 0)
++		return 0;
++
++	if (data == 0x8251)
++		return 0;
++
++	data = phy_read_mmd(phydev, MDIO_MMD_VEND1, 0x104);
++	if (data < 0)
++		return 0;
++
++	if ((data & 0xFFC0) != 0x1140)
++		return 0;
++
++	return 1;
++}
++
++static int rtl8261n_match_phy_device(struct phy_device *phydev,
++				     const struct phy_driver *phydrv)
++{
++	if (!rtlgen_is_c45_match(phydev, PHYID_RTL8261N, true))
++		return 0;
++
++	if (!phydev->mdio.bus->read_c45)
++		return 0;
++
++	int data = phy_read_mmd(phydev, MDIO_MMD_VEND1, 0x103);
++	if (data < 0)
++		return 0;
++
++	if (data == 0x8251)
++		return 0;
++
++	data = phy_read_mmd(phydev, MDIO_MMD_VEND1, 0x104);
++	if (data < 0)
++		return 0;
++
++	if ((data & 0xFFC0) == 0x1140)
++		return 0;
++
++	return 1;
++}
++
++static int rtl8264b_match_phy_device(struct phy_device *phydev,
++				     const struct phy_driver *phydrv)
++{
++	if (!rtlgen_is_c45_match(phydev, PHYID_RTL8264B, true))
++		return 0;
++
++	if (!phydev->mdio.bus->read_c45)
++		return 0;
++
++	int data = phy_read_mmd(phydev, MDIO_MMD_VEND1, 0x103);
++	if (data < 0)
++		return 0;
++
++	if (data == 0x8254)
++		return 0;
++
++	return 1;
++}
++
+ static int rtlgen_resume(struct phy_device *phydev)
+ {
+ 	int ret = genphy_resume(phydev);
+@@ -1705,6 +1822,394 @@ static irqreturn_t rtl8221b_handle_inter
+ 	return IRQ_HANDLED;
+ }
+ 
++static int rtl826x_probe(struct phy_device *phydev)
++{
++	struct device *dev = &phydev->mdio.dev;
++	struct rtl826x_priv *priv = NULL;
++
++	priv = devm_kzalloc(dev, sizeof(struct rtl826x_priv), GFP_KERNEL);
++	if (!priv)
++		return -ENOMEM;
++
++	memset(priv, 0, sizeof(struct rtl826x_priv));
++
++	priv->pnswap_tx = device_property_read_bool(dev, "realtek,pnswap-tx");
++	priv->pnswap_rx = device_property_read_bool(dev, "realtek,pnswap-rx");
++	priv->patch_rtk_serdes = device_property_read_bool(dev, "realtek,patch-rtk-serdes");
++	phydev->priv = priv;
++
++	return 0;
++}
++
++static int rtl8261n_probe(struct phy_device *phydev)
++{
++	struct rtl826x_priv *priv;
++	int ret;
++
++	ret = rtl826x_probe(phydev);
++	if (ret < 0)
++		return ret;
++
```

Comment:
The RTL8261N has a temperature sensor. You can activate it this way:
```suggestion
+
+      ret = rtl822x_probe(phydev);
+      if (ret < 0)
+              return ret;
+
```

~~~
root@OpenWrt:~# sensors
...
mt7530_0:05-mdio-5
Adapter: MDIO adapter
temp1:        +54.0°C  (high = +120.0°C)
...
~~~

It is likely that other PHYs added also have it built in.